pyproject.toml

```diff
@@ -525,4 +525,20 @@ dataframes = "rich"
 watcher_on_save = "autorun"
 output_max_bytes = 10_000_000
 std_stream_max_bytes = 2_000_000
-dotenv = [".env.testing"]
\ No newline at end of file
+dotenv = [".env.testing"]
+
+[tool.pixi.workspace]
+channels = ["conda-forge"]
+platforms = ["osx-arm64"]
+
+[tool.pixi.tasks]
+
+[tool.pixi.dependencies]
+nodejs = "20.*"
+pnpm = "9.*"
+hatch = ">=1.14.0,<2"
+make = ">=4.4.1,<5"
+pre_commit = ">=4.2.0,<5"
+uv = ">=0.6.12,<0.7"
+python = "3.12.*"
+pip = ">=25.0.1,<26"
```

Comment:
`ruff` and `typos` should be provided through hatch, but are also useful to have as dev tools to run standalone